### ISSUE TITLE
GUIDE Link

### ISSUE THREAD

**Issue by chr0m1ng** [coisadepro/Vostro-5470-Hackintosh#1]
Can you provide us your guide link on readme.md? I didn't found it... thanks

**Comment by mvsousa** [comment]
Hey, I'm new into hackintosh, but I have the same notebook model that you have, but I upgraded with a SSD. Do you know if its possible to make a dual boot with linux Ubuntu ? I would like to keep my Ubuntu for work.

**Comment by chr0m1ng** [comment]
Yes it is. Same as Windows, you just need to leave space for Mac, you can format the partition in HFS+ using gparted, just install hfs+ plugin, google it and you'll be fine

**Comment by pigdogkiki** [comment]
the shutdown patch do not work always，sometimes the applealc kexts do not work either.anybody can help?THANK YOU VERY MUCH!

**Comment by matheusfillipe** [comment]
what mac version are you guys even trying with?

**Comment by coisadepro** [comment]
Hey guys, sorry for not reading this earlier. I know it's been a long time since I took a look at this repo, but that's because this is not my main GitHub.

In the end I gave up on trying to make it run macOS, and I think you should do the same (or just accept the fact that the shutdown issue is unfixable).

My shutdown patch sometimes help, but it won't solve. The issue being a really screwed up BIOS firmware (UEFI firmware) that this laptop comes with.

I worked hard for about 4 months trying to get it to work, and managed to get it to 99%, except for the stupid shutdown issue that just can't be fixed. Even with my custom UEFI program that tries to do the trick,

How the issue works:
Whenever you shut the computer down from macOS, it won't turn back up unless you remove both the CMOS battery and the laptop battery (which requires you to disassemble it).

It doesn't matter if you force shutdown (press and hold the power button) or if the macOS panics and crashes. If the computer shuts down under macOS, it won't turn back on.

What I did notice that sometimes happened was that if instead of shutting it down, I just rebooted and opened the BIOS screen, and THEN shutdown, it would turn back on as expected. I don't know the reason. I think it might have to do with ACPI problems in the BIOS implementation itself (nothing to do with macOS or Hackintosh). I remember the same issue happening to the Windows 10 preview (before it was released).

What my patch does is simply that. It replaces the original shutdown implementation from macOS to a custom one that always reboots (instead of shutting down). Then, before botting the macOS (also before Clover), it checks if it was supposed to power off or reboot. If it should poweroff, it does that outside of macOS. If it was supposed to reboot, it calls Clover so that the boot can proceed.

That should work, right? Sometimes yes, but sometimes no. And then you'll have to disassemble it and do that boring proccess again and again.

So be prepared to be carrying around a screwdriver everywhere along with that laptop.

Let me also tell you something: that laptop's plastic is really fragile. I kept doing that for less than a month or something, and managed to break the ethernet (RJ-45) slot, lost some screws and a piece of plastic too.

Despite the issue with the ethernet slot, the laptop still works, but I just formatted and installed Windows. The Hackintosh experience on Desktop PCs is muuuuuch smoother than on laptops.

When looking for a laptop to do Hackintosh, always look for one that has no dedicated GPU, that has a Core i3 or superior, that has SATA drives (fancy things usually are harder or impossible to get to work), that the manufacturer is not Samsung (they use some display tech that don't play well with macOS).

Wi-fi won't work and can't be fixed. You'll need to replace it with something compatible. Check tonymacx86 or some other hackintosh forum for a list of alternatives. But make sure the laptop you're planning to buy has a Wi-fi card that can be replaced. Some are soldered, and then no Wi-fi for you.

Trackpad, keyboard and BIOS are kind of a bet. Best to check if anyone else has managed to do a successful hackintosh on that same machine.

That's it folks. If you want to keep trying, I wish you good luck. Be careful with the parts when disassembling and reassembling. It really is fragile.

The best version I got to work back in the day was Sierra. Altough High Sierra did work, the shutdown issue seemed to be more frequent.

**Comment by matheusfillipe** [comment]
Thanks for the reply @coisadepro! This laptop is really one of the worst to ever hackintosh, but was a nice experiment. This is my thread on tonymacx86 about it: https://www.tonymacx86.com/threads/vostro-5470.273654/ . 

After dissembling so many times  because of the shutdown problem the plastic cover got broken and the keyboard is quite useless and the screen is fliquering  when I move it . I just glued this all and installed ubuntu 20.04 on the entire disk. I use as a sort of a secondary monitor now.

Sadly I don't have a desktop to hackintosh but a qemu mac OS VM on my main linux machine is enough for me for now.

**Comment by pigdogkiki** [comment]
I also gave up this notebook, but thank you very much for your reply!
















At 2020-05-08 09:19:53, "CoisadePro" <notifications@github.com> wrote:

Hey guys, sorry for not reading this earlier. I know it's been a long time since I took a look at this repo, but that's because this is not my main GitHub.

In the end I gave up on trying to make it run macOS, and I think you should do the same (or just accept the fact that the shutdown issue is unfixable).

My shutdown patch sometimes help, but it won't solve. The issue being a really screwed up BIOS firmware (UEFI firmware) that this laptop comes with.

I worked hard for about 4 months trying to get it to work, and managed to get it to 99%, except for the stupid shutdown issue that just can't be fixed. Even with my custom UEFI program that tries to do the trick,

How the issue works:
Whenever you shut the computer down from macOS, it won't turn back up unless you remove both the CMOS battery and the laptop battery (which requires you to disassemble it).

It doesn't matter if you force shutdown (press and hold the power button) or if the macOS panics and crashes. If the computer shuts down under macOS, it won't turn back on.

What I did notice that sometimes happened was that if instead of shutting it down, I just rebooted and opened the BIOS screen, and THEN shutdown, it would turn back on as expected. I don't know the reason. I think it might have to do with ACPI problems in the BIOS implementation itself (nothing to do with macOS or Hackintosh). I remember the same issue happening to the Windows 10 preview (before it was released).

What my patch does is simply that. It replaces the original shutdown implementation from macOS to a custom one that always reboots (instead of shutting down). Then, before botting the macOS (also before Clover), it checks if it was supposed to power off or reboot. If it should poweroff, it does that outside of macOS. If it was supposed to reboot, it calls Clover so that the boot can proceed.

That should work, right? Sometimes yes, but sometimes no. And then you'll have to disassemble it and do that boring proccess again and again.

So be prepared to be carrying around a screwdriver everywhere along with that laptop.

Let me also tell you something: that laptop's plastic is really fragile. I kept doing that for less than a month or something, and managed to break the ethernet (RJ-45) slot, lost some screws and a piece of plastic too.

Despite the issue with the ethernet slot, the laptop still works, but I just formatted and installed Windows. The Hackintosh experience on Desktop PCs is muuuuuch smoother than on laptops.

When looking for a laptop to do Hackintosh, always look for one that has no dedicated GPU, that has a Core i3 or superior, that has SATA drives (fancy things usually are harder or impossible to get to work), that the manufacturer is not Samsung (they use some display tech that don't play well with macOS).

Wi-fi won't work and can't be fixed. You'll need to replace it with something compatible. Check tonymacx86 or some other hackintosh forum for a list of alternatives. But make sure the laptop you're planning to buy has a Wi-fi card that can be replaced. Some are soldered, and then no Wi-fi for you.

Trackpad, keyboard and BIOS are kind of a bet. Best to check if anyone else has managed to do a successful hackintosh on that same machine.

That's it folks. If you want to keep trying, I wish you good luck. Be careful with the parts when disassembling and reassembling. It really is fragile.

The best version I got to work back in the day was Sierra. Altough High Sierra did work, the shutdown issue seemed to be more frequent.

**Comment by droidzspy** [comment]
> Thanks for the reply @coisadepro! This laptop is really one of the worst to ever hackintosh, but was a nice experiment. This is my thread on tonymacx86 about it: https://www.tonymacx86.com/threads/vostro-5470.273654/ .
> 
> After dissembling so many times because of the shutdown problem the plastic cover got broken and the keyboard is quite useless and the screen is fliquering when I move it . I just glued this all and installed ubuntu 20.04 on the entire disk. I use as a sort of a secondary monitor now.
> 
> Sadly I don't have a desktop to hackintosh but a qemu mac OS VM on my main linux machine is enough for me for now.

I found a workaround for shutdown problem.
https://www.tonymacx86.com/threads/vostro-5470.273654/post-2127079
